### PR TITLE
Better document FiniteElement.

### PR DESCRIPTION
**Pull request by bangerth**
In particular, document the restriction_is_additive flags for which
there was basically nothing there so far. Also provide better documentation
about the constructor arguments.

### CODE REVIEW

**Comment by bangerth** [comment]
In partial reference to #1397

File: include/deal.II/fe/fe.h [hunk 1]
```diff
@@ -310,23 +310,142 @@ namespace hp
...
- * automatically, using FETools::compute_embedding_matrices().
+ * Once you have shape functions, you can define matrices that transfer
+ * data from one cell to its children or the other way around. This is
+ * a common operation in multigrid, of course, but is also used when
+ * interpolating the solution from one mesh to the one after mesh refinement,
+ * as well as in the definition of some error estimators.
  *
- * This can be achieved by
+ * To define the prolongation matrices, i.e., those matrices that
+ * are describe the transfer of a finite element field from one cell to
```
Comment (tamiko) [inline include/deal.II/fe/fe.h]: <i>that describe</i>

File: include/deal.II/fe/fe.h [hunk 2]
```diff
@@ -310,23 +310,142 @@ namespace hp
...
+ * by interpolation from the children. Rather, the only reasonable choice is to
+ * take it as the <i>average</i> value between the children -- so we are back
+ * to the sum operation, rather than the concatenation. Further thought shows that
+ * whether restriction should be additive or not is a property of the individual
+ * shape function, not of the finite element as a whole. Consequently, the
+ * FiniteElement::restriction_is_additive() function returns whether a
+ * particular shape function should act via concatenation (a return value
+ * of @p false) or via addition (return value of @p true), and the correct
+ * code for the overall operation is then as follows (and as, in fact,
+ * implement in DoFAccessor::get_interpolated_dof_values()):
```
Comment (tamiko) [inline include/deal.II/fe/fe.h]: <i>implemented</i>

**Comment by tamiko** [comment]
Please fix, rebase and merge.

**Pull request merged**